modules/core/include/opencv2/core/matx.hpp

```diff
@@ -99,13 +99,13 @@ In case if C++11 features are available, std::initializer_list can be also used
 template<typename _Tp, int m, int n> class Matx
 {
 public:
-    static const int       rows         = m;
-    static const int       cols         = n;
-    static const int       channels     = rows*cols;
-    static const int       shortdim     = (m < n ? m : n);
+    static const int        rows         = m;
+    static const int        cols         = n;
+    static const int        channels     = rows*cols;
+    static const int        shortdim     = (m < n ? m : n);
 #ifdef OPENCV_TRAITS_ENABLE_DEPRECATED
-    static const int       depth        = traits::Type<_Tp>::value;
```

Comment:
@alalek @hrnr This line sounds very strange for me! Should not it be `depth = traits::Depth<_Tp>::value` instead?

Reply:
`traits::Depth` should be here.
Only CUDA builds still used this deprecated code, probably it raises problems with Matx over `Vec3f` or similar.

BTW, we have plans to remove this deprecated section in master branch before 4.0 release.